src/choose-location/choose-location.component.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import { always } from "kremling";
+import { useConfig } from "@openmrs/esm-module-config";
+import { Trans } from "react-i18next";
+import {
+  getLoginLocations,
+  setSessionLocation
+} from "./choose-location.resource";
+import styles from "../styles.css";
+
+export default function ChooseLocation(props: ChooseLocationProps) {
+  const config = useConfig();
+  const [loginLocations, setLoginLocations]: [
+    Array<Object>,
+    (locations: Array<Object>) => void
+  ] = React.useState([]);
+  const [location, setLocation] = React.useState("");
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState("");
+  const locationInputRef = React.useRef<HTMLInputElement>(null);
+  const formRef = React.useRef<HTMLFormElement>(null);
+
+  React.useEffect(() => {
+    const sub = getLoginLocations().subscribe(locations => {
+      setLoginLocations(locations);
+    });
+
+    return () => sub.unsubscribe();
+  }, []);
+
+  React.useEffect(() => {
+    if (isSubmitting) {
+      setSessionLocation(location).then(data => {
+        navigate(props, config.links.loginSuccess);
+      });
```

Comment:
missing error handler. Omitting this will result in an unhandled promise rejection which will not be reported at all or very well to the user or to automated error tracking systems.

```suggestion
      }).catch(createErrorHandler())
```